### ISSUE TITLE
TOC is missing page numbers

### ISSUE THREAD

**Issue by kjambunathan** [kjambunathan/org-mode-ox-odt#175]
### Discussed in https://github.com/kjambunathan/org-mode-ox-odt/discussions/174

<div type='discussions-op-text'>

<sup>Originally posted by **PiotrPanasiuk** March 22, 2022</sup>
 When i export by ctrl x E and o. o my TOC is generated without page numbers and all entry are hyperlinks ( blue underlined). When I generate i.e. ctrl x E and r p TOC is generated with page numbers.
Do someone knows if there is possibility to create table of contents with page numbers and ox-odt?</div>

**Comment by kjambunathan** [comment]
Cc @PiotrPanasiuk, @QiangF

As you have noticed, the TOC (of headlines) doesn't generated page numbers.  I have converted the discussion in to an issue.  I will be committing a fix in another few minutes.

**Comment by kjambunathan** [comment]
To get this

![toc-with-pagenos](https://user-images.githubusercontent.com/19869864/159665308-a3db1fcf-3a87-4d19-a76b-84856a871154.png)


do this

```
#+options: ':nil *:t -:t ::t <:t H:3 \n:nil ^:t arch:headline
#+options: author:nil broken-links:mark c:nil creator:nil
#+options: d:(not "LOGBOOK") date:nil e:t email:nil f:t inline:t num:t
#+options: p:nil pri:nil prop:nil stat:t tags:t tasks:t tex:t
#+options: timestamp:t title:t toc:nil todo:t |:t
#+title: TOC with Page Nos.
#+date: <2022-03-23 Wed>
#+author: Jambunathan K
#+email: kjambunathan@gmail.com
#+language: en
#+select_tags: export
#+exclude_tags: noexport
#+creator: Emacs 29.0.50 (Org mode 9.5.2)
#+cite_export:

#+odt_preferred_output_format: png

#+odt_extra_styles: <style:style style:name="ReadmeBox"
#+odt_extra_styles:              style:parent-style-name="OrgTextBoxFrame"
#+odt_extra_styles:              style:family="graphic">
#+odt_extra_styles:   <style:graphic-properties draw:fill="solid"
#+odt_extra_styles:                             draw:fill-color="#ffff00"
#+odt_extra_styles:                             draw:wrap-influence-on-position="once-concurrent"
#+odt_extra_styles:                             fo:background-color="#ffff00"
#+odt_extra_styles:                             style:wrap="none" />
#+odt_extra_styles: </style:style>

#+ATTR_ODT: :width 15
#+ATTR_ODT: :style "ReadmeBox"
#+begin_textbox
If you are NOT seeing page numbers in Table of Contents, remember to
do one of the following:

- open the document in =LibreOffice= and do =Tools -> Update ->
   Update All=, or

- install the =LibreOffice Extension=
  =./contrib/odt/LibreOffice/OrgModeUtilities.oxt=, enable macro
  execution in =LibreOffice= documents and configure =Emacs= user
  option =org-odt-transform-processes= to include =Update All=.

Also see
[[https://github.com/kjambunathan/org-mode-ox-odt/issues/111#issuecomment-989495214][How
to make =LibreOffice= reload files on Export]]

#+end_textbox

#+toc: headlines

* Laboris incididunt exercitation mollit

Duis cillum eiusmod amet, duis ex nisi proident, laborum velit fugiat
reprehenderit dolore magna deserunt irure consectetur.  Non minim
aliqua fugiat nisi magna consequat adipiscing minim aute dolor
deserunt.

** Ipsum aliquip ipsum duis amet, nostrud

Consequat veniam, minim sit culpa laborum nostrud enim quis aute do
reprehenderit eu deserunt eu.

*** Velit veniam, veniam, reprehenderit excepteur

Consequat et exercitation officia fugiat est aliquip lorem dolor
excepteur qui elit, proident, officia amet, aliquip ex magna in
adipiscing occaecat.  Anim irure minim ipsum dolore labore ipsum
tempor cupidatat elit, labore.  Ullamco voluptate ad sint est nostrud
eiusmod ad pariatur sint officia mollit irure do sunt velit pariatur
excepteur in ipsum.


```

**Comment by kjambunathan** [comment]
@PiotrPanasiuk

(I hope you have seen the "Yellow" box)

The ODT exporter can't figure out the page nos.  So, you need  to rely on LibreOffice to fill in the page nos for you.  So, 

> If you are NOT seeing page numbers in Table of Contents, remember to do one of the following:
> - open the document in `LibreOffice` and do `Tools -> Update ->   Update All`, or
> - install the `LibreOffice Extension`  `./contrib/odt/LibreOffice/OrgModeUtilities.oxt`, enable macro  execution in `LibreOffice` documents and configure `Emacs` user  option `org-odt-transform-processes` to include `Update All`.

> Also see [How
to make =LibreOffice= reload files on Export](https://github.com/kjambunathan/org-mode-ox-odt/issues/111#issuecomment-989495214).

**Comment by kjambunathan** [comment]
The fix is available in [`elpa/ox-odt-9.5.2.405.tar`](https://github.com/kjambunathan/kjambunathan.github.io/tree/master/elpa).
